src/citation/views/citation_entry_view.py

```diff
@@ -43,23 +43,61 @@ def retrieve(self, request):
         )
 
     @action(detail=False, methods=["post"], permission_classes=[IsAuthenticated])
-    def pdf_uploads(self, request):
-        pdfs = request.FILES.getlist("pdfs[]")
-        created = []
-        for pdf in pdfs:
-            url = upload_to_s3(pdf, "citation_pdfs")
-            path = url.split(".com/")[1]
-            handle_creating_citation_entry.apply_async(
-                (
-                    path,
-                    request.user.id,
-                    request.data.get("organization_id"),
-                    request.data.get("project_id"),
-                ),
-                priority=5,
+    def upload_pdfs(self, request):
+        """
+        To enable in development:
+        1. Use Ngrok to create a tunnel on your backend port (usually 8000)
+        2. Go to staging-pdf-uploads-s3-trigger in AWS Lambda and look at code comments
+        """
+        if DEVELOPMENT:
+            raise Exception(
+                "See code comments to enable pdf uploads in dev environment"
             )
 
-        return Response({"created": created}, 200)
+        data = request.data
+        organization_id = data.get("organization_id")
+        project_id = data.get("project_id")
+        filename = data.get("filename")
+
+        cleaned_filename = clean_filename(f"{get_random_string(8)}_{filename}")
+        user_key = f"user_{request.user.id}"
+        boto3_session = session.Session()
+        s3_client = boto3_session.client("s3")
+        res = s3_client.generate_presigned_url(
+            "put_object",
+            Params={
+                "Bucket": AWS_STORAGE_BUCKET_NAME,
+                "Key": f"uploads/citation_pdfs/{user_key}_{cleaned_filename}",
+                "ContentType": "application/pdf",
+                "Metadata": {
+                    "x-amz-meta-created-by-id": f"{request.user.id}",
+                    "x-amz-meta-organization-id": f"{organization_id}",
+                    "x-amz-meta-project-id": f"{project_id}",
+                    "x-amz-meta-file-name": filename,
+                },
+            },
+            ExpiresIn=60 * 5,
+        )
+        return Response(res, status=200)
+
+    @action(detail=False, methods=["post"], permission_classes=[PDFUploadsS3CallBack])
+    def upload_pdfs_callback(self, request):
```

Comment:
is there a way to secure this endpoing so that the we only allow request from recognized sources? in this case S3

Reply:
Not sure if this is fail safe or how much logging we do at this level. But I just want to avoid this endpoint getting pinged from random sources

Reply:
This is a good observation. I have a permission class set to only allow incoming requests from our S3 trigger, so it can't be pinged from random sources (HTTP 403).